src/ast/passes/semantic_analyser.cpp

```diff
@@ -1240,7 +1240,8 @@ void SemanticAnalyser::visit(Call &call)
   } else if (call.func == "str") {
     auto &arg = call.vargs.at(0);
     const auto &t = arg.type();
-    if (!t.IsStringTy() && !t.IsIntegerTy() && !t.IsPtrTy()) {
+    if (!t.IsStringTy() && !t.IsIntegerTy() && !t.IsPtrTy() &&
+        (!t.IsArrayTy() || t.GetElementTy()->GetSize() != 1)) {
```

Comment:
Nit: I think we have this in multiple places, perhaps it's time to introduce something like
```cpp
bool SizedType::IsCString()
{
  return IsArrayTy() && GetElementTy()->IsIntegerTy() && GetElementTy()->GetSize() == 1;
}
```